### ISSUE TITLE
INTERNAL ERROR: Assertion failed: std::adjacent_find(mInputs.begin(), mInputs.end(), [](NetworkTensor* t1, NetworkTensor* t2) { return t1->getType() == t2->getType(); }) == mInputs.end()

### ISSUE THREAD

**Issue by HongchunSun** [NVIDIA/TensorRT#967]
## Description
I'm already got onnx model from pytorch model. When getting trt engine from onnx, get error as title. The error is caused by the pytorch conv2d layer, code as following
```
torch.nn.functional.conv2d(x, weight, self.bias, self.stride, self.padding, self.dilation, self.groups)
```
`x is torch.Size([1, 11, 1080, 1920]) torch.float32`
`weight is torch.Size([64, 11, 7, 7]) torch.float32`
`self.bias` is None
`self.stride = (2, 2)` 
` self.padding = (3, 3)`
`self.dilation = (1, 1)`
`self.groups = 1`

in onnx2trt.py trt network build code is
```
def create_trt_parser():
    TRT_LOGGER = trt.Logger(trt.Logger.WARNING)
    EXPLICIT_BATCH = 1 << (int)(trt.NetworkDefinitionCreationFlag.EXPLICIT_BATCH)
    EXPLICIT_PRE = 1 <<(int)(trt.NetworkDefinitionCreationFlag.EXPLICIT_PRECISION)
    flags = EXPLICIT_BATCH | EXPLICIT_PRE
    builder = trt.Builder(TRT_LOGGER)
    network = builder.create_network(flags)
    parser = trt.OnnxParser(network, TRT_LOGGER)
    return parser, builder, network
```
the error is 
```
[TensorRT] INTERNAL ERROR: Assertion failed: std::adjacent_find(mInputs.begin(), mInputs.end(), [](NetworkTensor* t1, NetworkTensor* t2) { return t1->getType() == t2->getType(); }) == mInputs.end()
../builder/Network.cpp:1505
Aborting...
[TensorRT] ERROR: ../builder/Network.cpp (1505) - Assertion Error in inferTensorPrecisionTypes: 0 (std::adjacent_find(mInputs.begin(), mInputs.end(), [](NetworkTensor* t1, NetworkTensor* t2) { return t1->getType() == t2->getType(); }) == mInputs.end())
```

## Environment

**TensorRT Version**: TensorRT-7.2.1.6.Ubuntu-18.04.x86_64-gun.cuda-10.2.cudnn8.0
**GPU Type**: Quadro RTX 6000
**Nvidia Driver Version**: 450.80.02
**CUDA Version**: 10.2.89
**CUDNN Version**: libcudnn8_8.0.4.30-1.deb
**Operating System + Version**: Ubuntu 18.04.5 LTS
**Python Version (if applicable)**: 3.7
**TensorFlow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 1.7.0
**Baremetal or Container (if container which image + tag)**:

**Comment by ttyio** [comment]
Hello @HongchunSun , thanks for reporting.
Did you using INT8 QAT in your model, if not could you remove the `trt.NetworkDefinitionCreationFlag.EXPLICIT_PRECISION` ?

**Comment by HongchunSun** [comment]
hello @ttyio, in my model, i'm used FLOAT32 QTA, but tensorrt prompt me to add parameter `trt.NetworkDefinitionCreationFlag.EXPLICIT_PRECISION`.

**Comment by DeathKnightChen** [comment]
@HongchunSun Hi, I met the same problem. May I ask have you solved this yet?

**Comment by ttyio** [comment]
Hello @HongchunSun , @DeathKnightChen , we have some issue in QAT support in TRT 7.2, could you share onnx mode that we could debug? thanks!

**Comment by DeathKnightChen** [comment]
@ttyio I have sent my onnx model to your email. Please check it. Many thanks.

**Comment by ttyio** [comment]
Thanks @DeathKnightChen , replied you in email.

Hello @HongchunSun , could you also send me your model? I guess you hit different issues because @DeathKnightChen has not using QAT in his model.

**Comment by HongchunSun** [comment]
> Thanks @DeathKnightChen , replied you in email.
> 
> Hello @HongchunSun , could you also send me your model? I guess you hit different issues because @DeathKnightChen has not using QAT in his model.

Hello @ttyio  @DeathKnightChen , I could not use onnx model. I used the pytorch model, and building the trt engine with custom plugin. I check the pytorch model training process, the using model is not using QAT. And error may be caused by the custom plugin. I will debug it continuously. If I am solving the problem, I will tell you.

**Comment by ttyio** [comment]
Hello @HongchunSun , could you remove `trt.NetworkDefinitionCreationFlag.EXPLICIT_PRECISION`  when QAT is not used, also TRT has no support for conv whose filter is also activation.

**Comment by HongchunSun** [comment]
hello @ttyio , I have tried to remove `trt.NetworkDefinitionCreationFlag.EXPLICIT_PRECISION`, but the tensorrt remind me to open it.

**Comment by ttyio** [comment]
Hello @HongchunSun , sorry this is a false alarm in TRT,  we have some issue in 7.2 , we thought this is convolution generated from QAT, but actually you are trying to make the filter of the conv from activation, right?

Current tensorrt has no support for conv filter from activation, if you really want to do this, you might need use the `refit` feature, we have an sample show how to use `refit` in 
https://github.com/NVIDIA/TensorRT/tree/master/samples/python/engine_refit_mnist

**Comment by HongchunSun** [comment]
Hello @ttyio , ok, I will check the filter of the conv, and read the sample which you provide carefully. Thank you.

**Comment by ttyio** [comment]
closing since no activity for more than 3 weeks, please reopen if you still have question, thanks!

**Comment by doomooo** [comment]
Hi, I meet this problem with ONNX QAT model in TensorRT7.2. Any suggestions? (BTW, TensorRT8 has no problem.) @ttyio
